.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,7 +1,26 @@
-## Description
+<!--
 
-_Please add a description of the changes that this PR introduces and the files that
-are the most critical to review._ 
+Please add a reference to the issue that this PR addresses and indicate which
+files are most critical to review. If it fully addresses a particular issue,
+please include "Closes #XXX" (where "XXX" is the issue number).
 
-Closes: #XXX
+If this PR is not associated with an issue, please ensure that it has a detailed
+description that describes the change and why you feel it's necessary. In such
+cases, especially if the work is not aligned with the team's current priorities,
```

Comment:
Done. Let me know if this seems better?

Reply:
Hmm, I think that small PRs for spelling / style / code massaging are fine without first seeking our input. Even if the PRs are actually sort of frivolous, I'd prefer we keep the barrier to that kind of thing low so that we can have continuous positive interactions with people in the open source ecosystem.

I think we should really just call out that, if a PR has a large scope whose impact is non-trivial, contributors should first create an issue and seek further discussion.

Reply:
I've added some language there now that I hope addresses this concern?